src/ui/components/results/alternativeverticalscomponent.js

```diff
@@ -96,9 +107,14 @@ export default class AlternativeVerticalsComponent extends Component {
    * @param {object} alternativeVerticals alternativeVerticals server response
    * @param {object} verticalsConfig the configuration to use
    */
-  static _buildVerticalSuggestions (alternativeVerticals, verticalsConfig) {
+  static _buildVerticalSuggestions (alternativeVerticals, verticalsConfig, context) {
     let verticals = [];
-    let queryParams = window.location.search;
+
```

Comment:
what do you think about refactoring these usages of SearchParams into to use addParamsToUrl in urlutils.js?

Reply:
If context is being added to persistent storage, I think it also shouldn't be necessary to explicitly add it to SearchParams. Correct me if I'm wrong, but the param should already be included in window.location.search.sbustring(1) right?

Reply:
edit: okay I read your commit message again and this is because these values are sometimes calculate before context is set. would building something like verticalSuggestions within setState instead of in the constructor fix some of these issues?

Reply:
* re SearchParams here: sure! what a handy function
* re SearchParams elsewhere: that feels dangerous to me and a bit out of scope of this particular item, would be down to refactor as part of another item
* re verticalSuggestions: I'm not sure I understand. The problem is that the URL is not updated at the time they are called. These functions are taking the URL from window.location.search. Building it in setState would still have the same timing issues, no?

Reply:
when setState is called the component's html updates so it should update the url

Reply:
Ah, as discussed offline, setContext is not necessarily called before all `addComponent`s, would need to check before every update:(